packages/muelu/src/Graph/Containers/MueLu_Aggregates_kokkos_def.hpp

```diff
@@ -180,6 +180,60 @@ namespace MueLu {
 
     return graph_;
   }
+  
+  template <class LocalOrdinal, class GlobalOrdinal, class DeviceType>
+  void 
+  Aggregates_kokkos<LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::ComputeNodesInAggregate(LO_view & aggPtr, LO_view & aggNodes, LO_view & unaggregated) const {
+    LO numAggs  = GetNumAggregates();
+    LO numNodes = vertex2AggId_->getLocalLength();
+    auto vertex2AggId = vertex2AggId_->getDeviceLocalView(Xpetra::Access::ReadOnly);
+    typename aggregates_sizes_type::const_type aggSizes = ComputeAggregateSizes(true);
+    LO INVALID = Teuchos::OrdinalTraits<LO>::invalid();
+
+    LO currNumUnaggregated=0;
+
+    // Construct the "rowptr" and the counter
+    // This is completely egregious to perform on host, but doesn't seem parallelizeable; is there a way to improve this?
+    typename LO_view::HostMirror aggPtr_h = Kokkos::create_mirror_view(aggPtr);
+    typename LO_view::HostMirror aggNodes_h = Kokkos::create_mirror_view(aggNodes);
+    typename LO_view::HostMirror unaggregated_h = Kokkos::create_mirror_view(unaggregated);
+    typename Kokkos::View<LO**,device_type>::HostMirror vertex2AggId_h = Kokkos::create_mirror_view(vertex2AggId); // TODO: this is a dualview, so we can't do it this way
+    typename aggregates_sizes_type::HostMirror aggSizes_h = Kokkos::create_mirror_view(aggSizes);
+    Kokkos::deep_copy(vertex2AggId_h, vertex2AggId); // TODO: same here
+    Kokkos::deep_copy(aggSizes_h, aggSizes);
+
+    Kokkos::resize(aggPtr_h,numAggs+1);
+    Kokkos::resize(aggNodes_h,numNodes);
+    Kokkos::resize(unaggregated_h,numNodes);
+    LO_view aggCurr("agg curr",numAggs+1);
+    typename LO_view::HostMirror aggCurr_h = Kokkos::create_mirror_view(aggCurr); // FIXME: not needed on device... Kokkos::View<LO*,host_type>?
+    Kokkos::deep_copy(aggCurr_h,aggCurr);
+
+    aggPtr_h(0) = 0; 
+    for(LO i=0; i<numAggs; i++) {
+      aggPtr_h(i+1) = aggSizes_h(i) + aggPtr_h(i);
+      aggCurr_h(i) = aggPtr_h(i);
+    }
+
+    // Stick the nodes in each aggregate's spot
+    for(LO i=0; i<numNodes; i++) {
```

Comment:
Same as above, can you rename `i` as `nodeIdx`?
Assuming you us a `parallel_scan` above, this should become a `parallel_for`

Reply:
Done!